src/index.ts

```diff
@@ -40,59 +61,63 @@ const isAddressAllowed = async (req: Request, res: Response) => {
             });
             return;
         }
-        const validAddresses = await contract.getAccountsList();
-        if (validAddresses instanceof Error) {
-            res.status(400).send({
-                error: `Couldn't get list of addresses to compare. ${validAddresses.message}`,
+        try {
+            const validAddresses = await cacheManager.get(CacheKeys.FULL_ALLOWED_LIST) as string[];
+            const address: string = req.query.address as string;
+            const isAllowed = validAddresses.indexOf(address) > -1;
+            res.send({
+                isAllowed,
             });
             return;
+        } catch (e) {
+            const err = e as Error;
+            console.log(`${err.name}, ${err.message}, ${err.stack}`);
+            res.status(400).send({ error: `Couldn't check validity of the address. ${err.message}` });
+            return;
         }
-        const address: string = req.query.address as string;
-        const isAllowed = validAddresses.indexOf(address) > -1;
-        res.send({
-            isAllowed,
-        });
+        
     } else {
         res.send({ isAllowed: true });
     }
 };
 
 const stargate = async (req: Request, res: Response) => {
-    const stargateAddress = await contract.getStargateAddress();
-    const tokenRegistry = await contract.getTokenRegistryAllowedList();
-    if (stargateAddress instanceof Error) {
-        res.status(400).send({ error: `Couldn't get stargate address. ${stargateAddress.message}` });
-        return;
-    }
-    if (tokenRegistry instanceof Error) {
-        res.status(400).send({ error: `Couldn't get assets. ${tokenRegistry.message}` });
-        return;
-    }
-    // TODO: Update config so node parses this env variable as a Boolean
-    // TODO: do we still need to distinguish between mainnet & devnet here?
-    if (CONFIG.API.mainnet === "TRUE") {
-        res.send({
-            current_address: stargateAddress,
-            ttl_expiry: new Date().setHours(24, 0, 0, 0),
-            ada: {
-                minLovelace: tokenRegistry.minLovelace,
-                fromADAFeeLovelace: "500000",
-                toADAFeeGWei: "500000"
-            },
-            assets: tokenRegistry.assets,
-        });
+
+    try {
+        const stargateAddress = await cacheManager.get(CacheKeys.STARGATE) as string;
+        const tokenRegistry = await cacheManager.get(CacheKeys.TOKEN_REGISTRY) as TokensRegistry;
+        
+        // TODO: Update config so node parses this env variable as a Boolean
+        // TODO: do we still need to distinguish between mainnet & devnet here?
```

Comment:
No need to differentiate

Reply:
Oh perfect, I wasn't sure about this one. Thanks! I'll remove that